Feat/create reveal modal

boranfurkan opened the pull request:
- Description:
- ClickUp links:
- What are the acceptance criteria? What are the steps to test that this code is working?

In this pull request I have completed the following task given in click up:
[](https://app.clickup.com/9015499537/inbox?tab=important)

I have implemented the component as like :
<img width="483" alt="Screenshot 2024-05-18 at 00 59 50" src="https://github.com/gator-labs/chomp/assets/96897046/77a6be0c-7dd7-4c8d-b1a2-e4392dbb3c7b">

I have used react-spring-bottom-sheet library as helper to speed up things. I have made local tests and seems working fine.

marvinmarnold reviewed (commented):
@boranfurkan this looks good. I'm waiting for confirmation about how the reveal amount will be determined. 

In the meantime, can you make the currency (BONK) and burn amount be defined in global constants that can be reused in other places?

marvinmarnold commented on app/providers/RevealProvider.tsx:
```diff
@@ -178,20 +181,25 @@ export function RevealContextProvider({ children }: { children: ReactNode }) {
+              </Button>
+            </div>
+            <p>
+              This will cost you <span className="font-bold">10000 BONK.</span>
```
Numbers should be displayed in this format `1,234,567.89`

boranfurkan commented:
I have created costs.ts file in constants folder and added reveal cost as constant there. I have also updated the places where I could find so far. So this pr now should be ready for merging sir @marvinmarnold.

boranfurkan commented:
UPDATE: Realised that I missed the formatting the currency. I have now created a function for that and implemented it. For your information @marvinmarnold.

AnteVuletic commented on app/utils/currency.ts:
```diff
@@ -1 +1,12 @@
 export const numberToCurrencyFormatter = new Intl.NumberFormat();
```
This formater should be deleted and new one should be applied, or this one should be used across the project. We don't need multiple currency formaters.

marvinmarnold replied:
@boranfurkan bumping this comment
